packages/smithy-http/src/smithy_http/aio/crt.py

```diff
@@ -174,13 +174,15 @@ def on_response(
                     kind=FieldPosition.HEADER,
                 )
 
-        self._response_future.set_result(
-            AWSCRTHTTPResponse(
-                status=status_code,
-                fields=fields,
-                body=self._body,
-            )
+        response = AWSCRTHTTPResponse(
+            status=status_code,
+            fields=fields,
+            body=self._body,
         )
+        if status_code != 200 and status_code >= 300:
```

Comment:
Technically the check for status_code != 200 is redundant with >= 300, but this is the logic the deserialize_error methods use.